dashboard/app/controllers/potential_teachers_controller.rb

```diff
@@ -4,13 +4,16 @@ class PotentialTeachersController < ApplicationController
 
   # url: /potential_teachers METHOD: POST
   def create
-    @potential_teacher = PotentialTeacher.new(potential_teacher_params)
-    begin
-      @potential_teacher.save!
-      render json: {message: 'Potential teacher created successfully'}, status: :created
-    rescue ActiveRecord::RecordInvalid => exception
-      raise "ERROR: #{exception.message}"
+    unless current_user
+      @potential_teacher = PotentialTeacher.new(potential_teacher_params)
+      begin
+        @potential_teacher.save!
+        render json: {message: 'Potential teacher created successfully'}, status: :created
+      rescue ActiveRecord::RecordInvalid => exception
```

Comment:
I know this wasn't added in this PR but just for my knowledge, why catch the exception instead of using the default?

Reply:
I'm not entirely sure! @juanmanzojr do you have thoughts here?